### PR TITLE
Adds 'clip' alias for clamp

### PR DESCRIPTION
**Pull request by mruberry**
Per title. Also updates our guidance for adding aliases to clarify interned_string and method_test requirements. The alias is tested by extending test_clamp to also test clip.

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -7034,80 +7034,85 @@ def test_logical_and(self, device):
...
+                res2[i] = max(min(res2[i], max_val), min_val)
+            self.assertEqual(torch.isnan(res1), torch.isnan(res2))
+
+            out = test_tens.clone()
+            op(test_tens, min=min_val, max=max_val, out=out)
+            self.assertEqual(torch.isnan(out), torch.isnan(res1))
+
+            res1 = op(test_tens, min=min_val)
+            res2 = test_tens.clone()
+            for i in iter_indices(res2):
```
Comment (ngimel) [inline test/test_torch.py:7095]: loop is not needed

File: test/test_torch.py [hunk 2]
```diff
@@ -7034,80 +7034,85 @@ def test_logical_and(self, device):
...
+                res2[i] = min(res2[i], max_val)
+            self.assertEqual(torch.isnan(res1), torch.isnan(res2))
+
+            op(test_tens, max=max_val, out=out)
+            self.assertEqual(torch.isnan(out), torch.isnan(res1))
+
+            error_msg = 'At least one of \'min\' or \'max\' must not be None'
+            with self.assertRaisesRegex(RuntimeError, error_msg):
+                method_op(m1)
```
Comment (ngimel) [inline test/test_torch.py:7113]: I don't see method_op tested anywhere other than here where it raises an error

File: test/test_torch.py [hunk 3]
```diff
@@ -7034,80 +7034,85 @@ def test_logical_and(self, device):
...
+            with self.assertRaisesRegex(RuntimeError, error_msg):
+                inplace_op(m1)
```
Comment (ngimel) [inline test/test_torch.py:7115]: Is nan propagation in clamp tested anywhere else? Vectorized and non-vectorized paths (could) propagate `nan` differently, so testing 1-element and 32-element tensors is needed to make sure that everything is ok.

File: torch/_torch_docs.py [hunk 4]
```diff
@@ -1497,6 +1496,12 @@ def merge_dicts(*dicts):
     tensor([ 0.5000, -0.4702, -0.4599,  0.5000])
 """.format(**common_args))
 
+add_docstr(torch.clip, r"""
+clip(input, min, max, *, out=None) -> Tensor
```
Comment (ngimel) [inline torch/_torch_docs.py:1500]: why is there "*" here but not in `clamp` declaration?

File: torch/testing/_internal/common_methods_invocations.py [hunk 5]
```diff
@@ -276,6 +276,8 @@ def method_tests():
         ('clamp', (), (None, 0.5), 'min_scalar', (True,)),
         ('clamp', (), (0.5, None), 'max_scalar', (True,)),
         ('clamp', (S, S), (), 'max_scalar_kwarg', (True,), (), (), ident, {'max': 1}),
+        ('clip', (S, S, S), dont_convert((0, 1)), '', (False,)),
+        ('clip_', (S, S, S), dont_convert((0, 1)), '', (False,)),
```
Comment (ngimel) [inline torch/testing/_internal/common_methods_invocations.py:280]: before your additions of `absolute_` and `clip_` looks like inplace versions weren't tested here, so maybe they shouldn't be?

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:280]: The alias test requires an inplace entry.

**Review by ngimel — APPROVED** [review]

**Comment by mruberry** [comment]
Note: per offline review, improvements to the existing clamp test will be separated and implemented when it's ported to the forthcoming test_unary_ufuncs.py.

**Pull request closed without merging**